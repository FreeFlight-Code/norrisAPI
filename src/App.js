import React from 'react';
import './App.scss';

function App() {
  return (
    <div className="App">
      Chuck Norris Rules!!!
      <div class="text-red">This will be red text.</div>
      <button className="blue">MORE INFO HEre</button>
      <button className="red">warning</button>
      <button className="green">Success</button>
      <button>default</button>
    </div>
  );
}

export default App;
