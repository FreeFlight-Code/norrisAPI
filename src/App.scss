@import './sass/_universal';
@import './sass/_buttons';
@import './sass/header';