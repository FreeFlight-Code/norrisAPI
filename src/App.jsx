import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Switch} from "react-router-dom";
import Header from './components/Header'
import Home from './pages/Home'
import Categories from './pages/Categories'
import Jokes from './pages/Jokes'
import Search from './pages/Search'

import './App.scss';

export default class App extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      loggedin: false,
      loading: false
    };
    this.loginUser = this.loginUser.bind(this);
  }

  loginUser(user){
    const {email, password} = user;
    renderModal();
    if(true){
      setTimeout(() => {
        console.log(`your email is ${email} and your password is ${password}.`)
        removeModal();
        this.setState({
          loggedin: true
        })
      }, 3000);
    }
  }

  toggleLoadingModal(){
    const loading = this.state;
    this.setState({
      loading: !loading
    })
  }

  render() {
    const {loggedin} = this.state;

    return (
      <div id="ChuckNorrisApp">
        <div id="modal"></div>
        <Header loggedin={loggedin} />

        <Router>
          {loggedin && (
            <Switch>
              <Route path="/categories">
                <Categories />
              </Route>
              <Route path="/jokes">
                <Jokes />
              </Route>
              <Route path="/search">
                <Search />
              </Route>
            </Switch>
          )}

          <Switch>
            <Route exact path="/">
              <Home loginUser={this.loginUser} />
            </Route>
            <Route path="/">
              <div>You must log in to see all content...</div>
            </Route>
          </Switch>
        </Router>
      </div>
    );
  }
}
 
function renderModal() {
  const element = (
    <div className="modal" id="loadingModal">
      <h1>Loading...</h1>
    </div>
  );
  ReactDOM.render(element, document.getElementById("modal"));
}
function removeModal() {
  const element = document.getElementById("modal")
  ReactDOM.unmountComponentAtNode(element);
}