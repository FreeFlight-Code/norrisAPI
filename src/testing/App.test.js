import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom')

describe('the App component',()=>{
	it("matches snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
