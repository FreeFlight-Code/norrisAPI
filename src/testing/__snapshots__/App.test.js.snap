// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="ChuckNorrisApp"
      >
        <header>
          <nav>
            <a
              href="/"
            >
              Home
            </a>
            <a
              href="/jokes"
            >
              Jokes
            </a>
            <a
              href="/categories"
            >
              Categories
            </a>
            <a
              href="/search"
            >
              Search
            </a>
          </nav>
          <button
            class="loggedin green"
          >
            Logged Out
          </button>
          <img
            alt="Chuck norris cartoon"
            src="https://assets.chucknorris.host/img/avatar/chuck-norris.png"
          />
        </header>
        <div
          class="page home"
        >
          <h3>
            Visit our pages above:
          </h3>
          <h4>
            Home Page allows you to log in.
          </h4>
          <h4>
            Jokes Page allows you to see the jokes you have viewed.
          </h4>
          <h4>
            Categories Page allows you to view the categories of jokes
          </h4>
          <h4>
            Search Page allows you to search jokes by key word.
          </h4>
          <form
            class="loginForm"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              data-testid="email-input"
              id="email"
              name="email"
              type="email"
              value=""
            />
            <span
              class="passwordContainer"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                type="checkbox"
              />
            </span>
            <input
              data-testid="password-input"
              id="password"
              name="password"
              type="password"
              value=""
            />
            <button
              class="submit blue glow disabled"
              data-testid="submit-input"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="ChuckNorrisApp"
    >
      <header>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/jokes"
          >
            Jokes
          </a>
          <a
            href="/categories"
          >
            Categories
          </a>
          <a
            href="/search"
          >
            Search
          </a>
        </nav>
        <button
          class="loggedin green"
        >
          Logged Out
        </button>
        <img
          alt="Chuck norris cartoon"
          src="https://assets.chucknorris.host/img/avatar/chuck-norris.png"
        />
      </header>
      <div
        class="page home"
      >
        <h3>
          Visit our pages above:
        </h3>
        <h4>
          Home Page allows you to log in.
        </h4>
        <h4>
          Jokes Page allows you to see the jokes you have viewed.
        </h4>
        <h4>
          Categories Page allows you to view the categories of jokes
        </h4>
        <h4>
          Search Page allows you to search jokes by key word.
        </h4>
        <form
          class="loginForm"
        >
          <label
            for="email"
          >
            Email
          </label>
          <input
            data-testid="email-input"
            id="email"
            name="email"
            type="email"
            value=""
          />
          <span
            class="passwordContainer"
          >
            <label
              for="password"
            >
              Password
            </label>
            <input
              type="checkbox"
            />
          </span>
          <input
            data-testid="password-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <button
            class="submit blue glow disabled"
            data-testid="submit-input"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
